packages/studio-base/src/panels/Plot/subscription.ts

```diff
@@ -0,0 +1,46 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import type { Immutable } from "@foxglove/studio";
+import type {
+  MessagePathPart,
+  RosPath,
+} from "@foxglove/studio-base/components/MessagePathSyntax/constants";
+import type { SubscribePayload } from "@foxglove/studio-base/players/types";
+
+const typeIsName = (part: Immutable<MessagePathPart>) => part.type === "name";
+
+export function pathToPayload(path: Immutable<RosPath>): SubscribePayload | undefined {
```

Comment:
```suggestion
export function pathToSubscribePayload(path: Immutable<RosPath>): SubscribePayload | undefined {
```

maybe rename the file to match?